Source/Core/Core/PowerPC/PPCSymbolDB.cpp

```diff
@@ -296,14 +291,37 @@ bool PPCSymbolDB::LoadMap(const std::string& filename, bool bad)
     if (section_name.empty())
       continue;
 
-    // Detect two columns with three columns fallback
+    // Column detection heuristic
     if (column_count == 0)
     {
-      const std::string_view stripped_line = StripSpaces(line);
-      if (std::count(stripped_line.begin(), stripped_line.end(), ' ') == 1)
-        column_count = 2;
-      else
+      constexpr auto is_hex_str = [](const std::string& s) {
+        return !s.empty() && s.find_first_not_of("0123456789abcdefABCDEF") == std::string::npos;
+      };
+      const std::string stripped_line(StripSpaces(line));
+      std::istringstream oss(stripped_line);
+      std::string word;
+
+      // Two columns format:
+      // 80004000 zz_80004000_
+      if (!(oss >> word) || word.length() != 8 || !is_hex_str(word))
```

Comment:
Did you consider `isxdigit` coupled with `std::all_of` ?  Not sure if there are any advantages to that.

Reply:
I did and I need to wrap `isxdigit(int)` around a lambda with an unsigned cast to avoid an assert error (`char` might be signed and negative) or use the locale version of it. However, I don't know if `isxdigit` is faster or not.